File: portScan.rb
#1/usr/local/bin/ruby

require 'socket'

puts "What ip: "
ip = gets
puts "What port: "
port = gets
ip = ip.chomp
port = port.chomp

begin
    socket = TCPSocket.new(ip, port)
    status = "open"
rescue Errno::ECONNREFUSED, Errno::ETIMEDOUT
    status = "closed"
end
puts "Port #{port} is #{status}."

File: portScan.py
#! /usr/bin/python3

import socket

ip = input("Enter the ip: ")
port = input ("Enter the port: ")
port = int(port)

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
if s.connect_ex((ip, port)):
    print("Port ", port, " is closed")
else:
    print("Port ", port, " is open")
